### PR TITLE
Implement kubectl wrapper for testrunner

### PR DESCRIPTION
**Pull request by chentex**
## Why is this PR needed?

adds an kubectl wrapper to help with the execution of e2e testing in testrunner.

Fixes SUSE/avant-garde#550

## What does this PR do?

adds an kubectl wrapper to help with the execution of e2e testing in testrunner.
also contains a sample test that will deploy a nginx and test it.

### CODE REVIEW

**Comment by chentex** [comment]
It is replaced by #496

File: ci/infra/testrunner/tests/test_deployments.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+from platforms import Platform
+from skuba import Skuba
+from kubectl import Kubectl
+import pytest
+import time
+from timeout_decorator import timeout
+
+@pytest.fixture
```
Comment (pablochacin) [inline ci/infra/testrunner/tests/test_deployments.py]: This fixture should be moved to the test library `tests/conftest.py`and re-used in both the deployment and worker tests. Can you add a commit to do so?

File: ci/infra/testrunner/kubectl/kubectl.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    def __init__(self, conf, platform):
+        self.conf = conf
+        self.utils = Utils(self.conf)
+        self.platform = Platform.get_platform(conf, platform)
+
+
+    def create_deployment(self, name, image):
+        print("create a new deployment {}".format(name))
+        try:
+            self._run_kubectl("create deployment {name}  --image={image}"
```
Comment (MaximilianMeister) [inline ci/infra/testrunner/kubectl/kubectl.py]: MINOR: there's some unnecessary whitespace

File: ci/infra/testrunner/kubectl/kubectl.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        try:
+            return self.utils.runshellcommand_withoutput("curl {ip}:{port}{path}"
+                                                        .format(ip=ip_address[worker],port=port,path=path), False)
+        except Exception as ex:
+            raise Exception("Error executing cmd {}") from ex
+
+
+    def _run_kubectl(self, command):
+        try:
+            return self.utils.runshellcommand_withoutput("kubectl --kubeconfig={cwd}/test-cluster/admin.conf -o json {command}"
```
Comment (MaximilianMeister) [inline ci/infra/testrunner/kubectl/kubectl.py]: i guess the `testrunner` creates the `test-cluster` folder, maybe that could be passed in in case the folder name should change at some point

File: ci/infra/testrunner/tests/test_deployments.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
+from platforms import Platform
+from skuba import Skuba
+from kubectl import Kubectl
+import pytest
+import time
+from timeout_decorator import timeout
+
+def test_nginx_deployment(setup, skuba, kubectl):
```
Comment (MaximilianMeister) [inline ci/infra/testrunner/tests/test_deployments.py]: are you also planning on having a jenkins pipeline as a consumer of this test?

Comment (chentex) [reply ci/infra/testrunner/tests/test_deployments.py]: The idea is to have one nightly that runs all of this test scenarios

File: ci/infra/testrunner/kubectl/kubectl.py (hunk 3 above)
Comment (pablochacin) [reply ci/infra/testrunner/kubectl/kubectl.py]: Yes, we have this name hard-coded in other places. I suggest another PR to fix this in all the testruner.
